source/_posts/2017-11-29-hassio-virtual-machine.markdown

```diff
@@ -0,0 +1,151 @@
+root@hassio:/home/ha# apt-key add gpg
+OK
+root@hassio:/home/ha# echo "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" | tee -a /etc/apt/sources.list.d/docker.list
+root@hassio:/home/ha# apt-get update
+```
+
+Now, it's possible to install a current release of [Docker](https://www.docker.com/).
+
+```bash
+root@hassio:/home/ha# apt-get -y install docker-ce
+```
+
+Start `docker` and enable it.
+
+```bash
+root@hassio:/home/ha# systemctl start docker && systemctl enable docker
+```
+
+An [installation script](https://github.com/home-assistant/hassio-build/tree/master/install#install-hassio) will take care about the setup of all moving parts.
+
+```bash
+root@hassio:/home/ha# curl -sL https://raw.githubusercontent.com/home-assistant/hassio-build/master/install/hassio_install | bash -
+[INFO] Install supervisor docker
+[INFO] Install generic HostControl
+[INFO] Install startup scripts
+[INFO] Init systemd
+Created symlink /etc/systemd/system/multi-user.target.wants/hassio-supervisor.service → /etc/systemd/system/hassio-supervisor.service.
+[INFO] Start services
+```
+
+If it's done, then there will be two new containers.
+
+```bash
+root@hassio:/home/ha# docker ps
+CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS              PORTS               NAMES
+ada5bbfc74f0        homeassistant/qemux86-64-homeassistant   "/usr/bin/entry.sh..."   4 minutes ago       Up 4 minutes                            homeassistant
+5954ac452ffc        homeassistant/amd64-hassio-supervisor    "/usr/bin/entry.sh..."   7 minutes ago       Up 7 minutes                            hassio_supervisor
+```
+
+After a connection to the container which is containing Home Assistant is made, you will see the log output.
+
+```bash
+root@hassio:/home/ha# docker attach --sig-proxy=false ada5bbfc74f0
+2017-11-28 19:24:30 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=sun.sun, old_state=<state sun.sun=below_horizon; next_dawn=2017-11-29T06:17:58+00:00,...
+```
+
+For further details about the container, `inspect` can help.
+
+```bash
+root@hassio:/home/ha# docker inspect bb32b525d1ad
+[...]
+            "OnBuild": null,
+            "Labels": {
+                "io.hass.arch": "amd64",
+                "io.hass.machine": "qemux86-64",
+                "io.hass.type": "homeassistant",
+                "io.hass.version": "0.58.1"
+            }
+[...]
+```
+
+Hass.io is now ready. The frontend is available at [http://192.168.0.109:8123](http://192.168.0.109:8123). Yes, the IP address is the one of the guest.
+
+<p class='img'>
+  <img src='/images/blog/2017-11-hassio-virtual/hassio.png' />
+  Hass.io overview
+</p>
+
+Keep in mind that there are limitations with this approach. Not all [add-ons](/addons/), especially the [SSH add-on](/addons/ssh/), will work and some don't make sense to use as the hardware is not present.
```

Comment:
Suggestion: The Community add-on version of SSH will work (https://github.com/hassio-addons/addon-ssh).

Reply:
Nice. I didn't know that.

Reply:
What work not with core ssh add-on?